cmd/boost/retrieve_piece_cmd.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+
+	bcli "github.com/filecoin-project/boost/cli"
+	clinode "github.com/filecoin-project/boost/cli/node"
+	"github.com/filecoin-project/boost/cmd"
+	"github.com/filecoin-project/boost/retrievalmarket/lp2pimpl"
+	"github.com/filecoin-project/boost/retrievalmarket/types"
+	"github.com/filecoin-project/go-address"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/ipfs/go-cid"
+	"github.com/urfave/cli/v2"
+)
+
+var retrievePieceCmd = &cli.Command{
+	Name:  "retrieve-piece",
```

Comment:
I'd suggest adding a `retrieve` category, and renaming this action to `query-ask`, so the CLI command would be
`boost retrieve query-ask --provider=t01234 --piece-cid=1234`

I also don't think we really need to provide the piece cid. The path to get things can be "well-known", eg `/piece/<cid>.car`